src/SqlAsyncCollector.cs

```diff
@@ -91,6 +92,13 @@ public SqlAsyncCollector(IConfiguration configuration, SqlAttribute attribute, I
             this._attribute = attribute ?? throw new ArgumentNullException(nameof(attribute));
             this._logger = logger;
             TelemetryInstance.TrackCreate(CreateType.SqlAsyncCollector);
+            using (SqlConnection connection = BuildConnection(attribute.ConnectionStringSetting, configuration))
```

Comment:
I would have liked to avoid doing this here for perf reasons (avoid opening a new connection if possible) but we don't open the connection until the rows are flushed which may be well after the function actually starts running. But something to consider moving if we think that's not worth the extra overhead.